src/main/java/com/splunk/hecclient/LoadBalancer.java

```diff
@@ -66,4 +116,80 @@ public void send(final EventBatch batch) {
+        log.debug("Running healthcheck at an interval of {} seconds", this.hecConfig.getlbPollInterval()/1000.0);
+            Enumeration<String> iter = indexerInfo.keys();
+            for (; iter.hasMoreElements();) {
+                String indexerToCheck = iter.nextElement();
+                try {
+                    indexerCheckResult(sendHealthCheckRequest(indexerToCheck), indexerToCheck);
+                } catch (Exception e) {
+                    indexerCheckResult(false, indexerToCheck);
+                    log.error("encountered exception when checking health of indexer {}, " +
+                            "this means indexer and its channels are removed from the loadbalancer", e.getMessage());
+                }
+            }
+
+    }
+
+    private synchronized void indexerCheckResult(boolean success, String indexer) {
+        log.debug("indexerInfo map has {} ", indexerInfo);
+        log.debug("discardedIndexers map has {} ", discardedIndexers);
+        log.debug("Channels List is : {}", channels);
+
+        if(success) {
+            if(discardedIndexers.contains(indexer)) {
+                log.info("healthcheck passed for {} indexer, adding this indexer and its channels to the loadbalancer, " +
+                        "this indexer was previously removed from loadbalancer", indexer);
+                // Add channels for this indexer as healthcheck passed
+                List<HecChannel> channelsToAdd = indexerInfo.get(indexer);
+                channelsToAdd.forEach((channel) -> channels.add(channel));
+                discardedIndexers.remove(indexer);
+                index = 0;
+            }
+        } else {
+            if(!discardedIndexers.contains(indexer)) {
+                log.info("healthcheck failed for {} indexer, removing this indexer and its channels from the loadbalancer", indexer);
+                // Remove channels for this indexer as healthcheck failed
+                List<HecChannel> channelsToRemove = indexerInfo.get(indexer);
+                channelsToRemove.forEach((channel) -> remove(channel));
+                discardedIndexers.add(indexer);
+                index = 0;
+            }
+        }
+    }
+
+    public boolean sendHealthCheckRequest(String indexerUrl) throws IOException {
+        boolean retVal = false;
+        String url = indexerUrl + healthCheckEndpoint;
+        final HttpGet httpGet = new HttpGet(url);
+        httpGet.setHeaders(headers);
+        int status = -1;
+        CloseableHttpResponse resp = httpClient.execute(httpGet, context);
+        try {
+            status = resp.getStatusLine().getStatusCode();
+        } finally {
+            resp.close();
+            if(status == 200) {
+                retVal = true;
+            } else if (status == 400) {
+                log.info("Invalid HEC token for indexer {}", url);
+            } else if (status == 503) {
+                log.info("HEC is unhealthy, queues are full for indexer {}", url);
+            }
```

Comment:
should we add a generic log for status code other than 200, 400 and 503.

Reply:
we provide an INFO log when we add or remove indexers, otherwise it will too noisy